src/creatures/players/player.cpp

```diff
@@ -7280,23 +7282,53 @@ void Player::sendUnjustifiedPoints() const {
 			}
 		}
 
-		const bool isRed = getSkull() == SKULL_RED;
+		const bool isRedOrBlack = getSkull() == SKULL_RED || getSkull() == SKULL_BLACK;
 
-		auto dayMax = ((isRed ? 2 : 1) * g_configManager().getNumber(DAY_KILLS_TO_RED));
-		auto weekMax = ((isRed ? 2 : 1) * g_configManager().getNumber(WEEK_KILLS_TO_RED));
-		auto monthMax = ((isRed ? 2 : 1) * g_configManager().getNumber(MONTH_KILLS_TO_RED));
+		auto dayMax = ((isRedOrBlack ? 2 : 1) * g_configManager().getNumber(DAY_KILLS_TO_RED));
+		auto weekMax = ((isRedOrBlack ? 2 : 1) * g_configManager().getNumber(WEEK_KILLS_TO_RED));
+		auto monthMax = ((isRedOrBlack ? 2 : 1) * g_configManager().getNumber(MONTH_KILLS_TO_RED));
 
 		const uint8_t dayProgress = std::min(std::round(dayKills / dayMax * 100), 100.0);
 		const uint8_t weekProgress = std::min(std::round(weekKills / weekMax * 100), 100.0);
 		const uint8_t monthProgress = std::min(std::round(monthKills / monthMax * 100), 100.0);
 		uint8_t skullDuration = 0;
-		if (skullTicks != 0) {
-			skullDuration = std::floor<uint8_t>(skullTicks / (24 * 60 * 60 * 1000));
+		//  If player is still redskull or blackskull but getSkullTicks is 0, calculate time left from last kill
+		if (getSkull() == SKULL_RED || getSkull() == SKULL_BLACK) {
+			auto &db = Database::getInstance();
+			std::ostringstream query;
+			query << "SELECT `time` FROM `player_kills` WHERE `player_id` = " << guid << " ORDER BY `time` DESC LIMIT 1;";
+			DBResult_ptr result = db.storeQuery(query.str());
+			int64_t lastKillTime = 0;
+			if (result && result->hasNext()) {
+				lastKillTime = std::stoll(result->getString("time"));
```

Comment:
Good point on retrieving `time`. Using `std::stoll` here is risky because it throws exceptions for invalid or out-of-range values, which can cause the server to crash. A safer alternative is `std::from_chars`, which avoids exceptions and gives explicit error codes:

```suggestion
				const std::string& timeStr = result->getString("time");
				auto [ptr, ec] = std::from_chars(timeStr.data(),
                                 timeStr.data() + timeStr.size(),
                                 lastKillTime);
				if (ec != std::errc()) {
    				// handle parse error, e.g. log or fallback
    				lastKillTime = 0;
				}
```

This keeps the code exception-free and more robust against malformed DB values.